src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestsTest.kt

```diff
@@ -0,0 +1,180 @@
+        createTestBuilder().use { testBuilder ->
+            val createdVacation = testBuilder.manager.vacationRequests.createVacationRequests(testVacationRequest)
+            val vacations = testBuilder.manager.vacationRequests.findVacationRequests(createdVacation.id!!)
+
+            assertEquals(1, vacations.person)
+            assertEquals(LocalDate.now().toString(), vacations.startDate)
+            assertEquals(LocalDate.now().plusDays(1).toString(), vacations.endDate)
+            assertEquals(2, vacations.days)
+            assertEquals(VacationType.VACATION, vacations.type)
+            assertEquals("Lomaa!!!", vacations.message)
+            assertEquals(VacationRequestStatus.PENDING, vacations.projectManagerStatus)
+            assertEquals(VacationRequestStatus.PENDING, vacations.hrManagerStatus)
+            assertEquals(UUID.fromString("50bd84bc-d7f7-445f-b98c-4f6a5d27fb55"), vacations.createdBy)
+            assertEquals(UUID.fromString("50bd84bc-d7f7-445f-b98c-4f6a5d27fb55"), vacations.lastUpdatedBy)
+        }
+    }
+
+    /**
+     * Tests /v1/listVacationRequest -endpoint
+     */
+    @Test
+    fun testListVacationRequests() {
+        createTestBuilder().use { testBuilder ->
+            testBuilder.manager.vacationRequests.createVacationRequests(testVacationRequest)
+
+            val vacations = testBuilder.manager.vacationRequests.listVacationRequests(personId = 1)
+
+            assertEquals(1, vacations[0].person)
+            assertEquals(LocalDate.now().toString(), vacations[0].startDate)
+            assertEquals(LocalDate.now().plusDays(1).toString(), vacations[0].endDate)
+            assertEquals(2, vacations[0].days)
+            assertEquals(VacationType.VACATION, vacations[0].type)
+            assertEquals("Lomaa!!!", vacations[0].message)
+            assertEquals(VacationRequestStatus.PENDING, vacations[0].projectManagerStatus)
+            assertEquals(VacationRequestStatus.PENDING, vacations[0].hrManagerStatus)
+            assertEquals(UUID.fromString("50bd84bc-d7f7-445f-b98c-4f6a5d27fb55"), vacations[0].createdBy)
+            assertEquals(UUID.fromString("50bd84bc-d7f7-445f-b98c-4f6a5d27fb55"), vacations[0].lastUpdatedBy)
+        }
+    }
+
+    /**
+     * Tests /v1/updateVacationRequest -endpoint
+     */
+    @Test
+    fun testUpdateVacationRequests() {
+        createTestBuilder().use { testBuilder ->
+            testBuilder.userA.vacationRequests.createVacationRequests(testVacationRequest)
+            val vacations1 = testBuilder.manager.vacationRequests.listVacationRequests(personId = 1)
+
+            assertEquals(VacationRequestStatus.PENDING, vacations1[0].projectManagerStatus)
+            assertEquals(UUID.fromString("7276979e-2f08-4d52-9541-0d10aa3806fe"), vacations1[0].lastUpdatedBy)
+
+            testBuilder.manager.vacationRequests.updateVacationRequests(
+                id = vacations1[0].id!!,
+                vacationRequest = VacationRequest(
```

Comment:
Doesn't stop this PR from being approved but as a general tip related to Kotlin; data classes have a `copy` method that returns a copy of the given object with changed parameters. 
E.g. `testVacationRequest.copy(projectManagerStatus = VacationRequestStatus.APPROVED)´ would offer a copy of the `testVacationRequest` object, but with reassigned `projectManagerStatus`.